### PR TITLE
Skip missing cards/dashboards in exposures

### LINKED ISSUE #237: Exposure Extraction Error | In Case of Missing Cards in collection
**Context:** 

When I use Metabase API to delete archived cards, it creates inconsistency because card are still referenced in collections.
So [at this step](https://github.com/gouline/dbt-metabase/blob/9fc8bcd67e6f418bb45561c238c8ffa6dd9dfb49/dbtmetabase/_exposures.py#L98C6-L98C68), it's possible to search a deleted card/dashboard causing an HTTP code 400 error and makes application crash.

**Problem**

In case of inconsistency between collections and cards/dashboards, dbt-metabase will crash because it tries to query a missing card/dashboard. 

**Request**

Skip a card/dashboard which is referenced in a collection but has been deleted.

**Technical Solution**

Test for each items in a Collection if the associated resource is reachable otherwise continue to the next item.
We could add a try&catch statement for get_dashboard and get_card as we did [here](https://github.com/gouline/dbt-metabase/blob/9fc8bcd67e6f418bb45561c238c8ffa6dd9dfb49/dbtmetabase/metabase.py#L166)

**Note**

Event if it's not the purpose of that tool, it could be interresting to highlight inconsistency with the logger in case of unreachable item resource.

### PR DESCRIPTION
**Pull request by dosnep**
- Fix: dbt-metabase now skips unreachable resources {card, dashboard, user} in case of **requests.exceptions.HTTPError: 404 Client Error: Not Found for url**

Closes #237

### CODE REVIEW

File: dbtmetabase/metabase.py [hunk 1]
```diff
@@ -145,15 +145,27 @@ def get_collection_items(
 
     def get_card(self, uid: str) -> Mapping:
```
Comment (gouline) [inline dbtmetabase/metabase.py:146]:
```suggestion
    def find_card(self, uid: str) -> Optional[Mapping]:
```

File: dbtmetabase/metabase.py [hunk 2]
```diff
@@ -145,15 +145,27 @@ def get_collection_items(
...
+            if error.response.status_code == 404:
+                _logger.warning("Card '%s' not found", uid)
+                return None
+            raise
 
     def format_card_url(self, uid: str) -> str:
         """Formats URL link to a card (known as question in Metabase UI)."""
         return f"{self.url}/card/{uid}"
 
     def get_dashboard(self, uid: str) -> Mapping:
```
Comment (gouline) [inline dbtmetabase/metabase.py:160]:
```suggestion
    def find_dashboard(self, uid: str) -> Optional[Mapping]:
```

**Review by gouline — CHANGES_REQUESTED** [review]
Please read [CONTRIBUTING.md](https://github.com/gouline/dbt-metabase/blob/master/CONTRIBUTING.md#pull-requests) for local checks to run before committing, to avoid waiting for failed CI.

**Comment by dosnep** [comment]
> ommitting, to avoid waiting for failed CI.

CI has been tested locally (fmt-check and unit tests)

**Comment by gouline** [comment]
Please read [CONTRIBUTING.md](https://github.com/gouline/dbt-metabase/blob/master/CONTRIBUTING.md#pull-requests) **carefully**. You only ran the formatting check, you have to run all of them.

**Pull request merged**